Solutions/Ais.Net.Receiver/IStorageClient.cs

```diff
@@ -0,0 +1,14 @@
+﻿// <copyright file="Program.cs" company="Endjin">
+// Copyright (c) Endjin. All rights reserved.
+// </copyright>
+
+namespace Ais.Net.Receiver
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public interface IStorageClient
+    {
+        Task PersistAsync(IEnumerable<string> messages);
```

Comment:
Actually, having read through the rest of the code, I now wonder if this should be `ICollection<string>`, to better reflect the fact that your only implementation of this that it expects the relevant data to be available immediately. The nature of this particular method is that it doesn't really want the source data to be lazy. (And you're feeding it with the output of the `Buffer` operator, which, IIRC, actually supplies you with an `IList<string>`, because by definition, `Buffer` only supplies you with fully-populated buffers.)